src/isomorphic/classic/types/__tests__/ReactPropTypes-test.js

```diff
@@ -40,7 +43,9 @@ function typeCheckPass(declaration, value) {
     props,
     'testProp',
     'testComponent',
-    ReactPropTypeLocations.prop
+    ReactPropTypeLocations.prop,
+    null,
+    ReactPropTypesSecret
```

Comment:
I’m a bit concerned about this. We’re not actually testing that *React* is passing the secret when validating. So if React doesn’t do it in some case, we’ll miss this.

Can we change these `typeCheckFail` and `typeCheckPass` to be more like integration tests? i.e. they would create a class and then make sure that calling `createElement` produces (or doesn’t produce) the relevant warning. Or am I missing something?